# BRIEF PREDICTION DE LA GRAVITE DES ACCIDENTS DE LA ROUTE

## Import des librairies

In [41]:
import numpy as np
import pandas as pd

import plotly.express as px
import matplotlib.pyplot as plt
import seaborn as sns

## Véhicules

### Import du jeu de données 

In [46]:
df_vehicules = pd.read_csv("vehicules-2023.csv", delimiter=';',encoding='utf-8') 
df_vehicules

Num_Acc  id_vehicule num_veh  senc  catv  obs  obsm  choc  manv  \
0      202300000001  155 680 557     A01     1    30    0     0     5     1   
1      202300000002  155 680 556     A01     2     7    0     1     1     1   
2      202300000003  155 680 554     B01     1     2    0     2     1    16   
3      202300000003  155 680 555     A01     2     7    0     2     2    15   
4      202300000004  155 680 551     B01     1     7    0     2     9     2   
...             ...          ...     ...   ...   ...  ...   ...   ...   ...   
93580  202300054820  155 583 348     A01     3     7    0     9     4    26   
93581  202300054821  155 583 346     A01     1    30    0     2     3     9   
93582  202300054821  155 583 347     B01     1     7    0     2     2     9   
93583  202300054822  155 583 344     A01     2     7    0     2     7    22   
93584  202300054822  155 583 345     B01     2    50   14     2     1     1   

       motor  occutc  
0          1     NaN  
1          1     NaN  
2          1     NaN  
3          1     NaN  
4          4     NaN  
...      ...     ...  
93580      1     NaN  
93581      1     NaN  
93582      1     NaN  
93583      1     NaN  
93584      3     NaN  

[93585 rows x 11 columns]

In [50]:
df_vehicules.head(20)

Num_Acc  id_vehicule num_veh  senc  catv  obs  obsm  choc  manv  \
0   202300000001  155 680 557     A01     1    30    0     0     5     1   
1   202300000002  155 680 556     A01     2     7    0     1     1     1   
2   202300000003  155 680 554     B01     1     2    0     2     1    16   
3   202300000003  155 680 555     A01     2     7    0     2     2    15   
4   202300000004  155 680 551     B01     1     7    0     2     9     2   
5   202300000004  155 680 552     C01     1     7    0     2     4     0   
6   202300000004  155 680 553     A01     1    10    0     2     1     2   
7   202300000005  155 680 549     B01     1     7    0     2     2     1   
8   202300000005  155 680 550     A01     1     7    0     2     3     1   
9   202300000006  155 680 548     A01     1     7    0     1     3    15   
10  202300000007  155 680 545     C01     2    10    0     2     1     1   
11  202300000007  155 680 546     A01     2     7    0     2     4    23   
12  202300000007  155 680 547     B01     2     7    0     2     1     1   
13  202300000008  155 680 543     B01     2     7    0     2     1     1   
14  202300000008  155 680 544     A01     3     7    0     2     2     1   
15  202300000009  155 680 541     A01     2     7    0     2     2    15   
16  202300000009  155 680 542     B01     1    32    0     2     1     1   
17  202300000010  155 680 540     B01     1     7    0     1     1     1   
18  202300000011  155 680 538     A01     2     7    0     2     5    26   
19  202300000011  155 680 539     B01     2     7    0     2     2     1   

    motor  occutc  
0       1     NaN  
1       1     NaN  
2       1     NaN  
3       1     NaN  
4       4     NaN  
5       4     NaN  
6       1     NaN  
7       1     NaN  
8       1     NaN  
9       1     NaN  
10      0     NaN  
11      2     NaN  
12      1     NaN  
13      1     NaN  
14      2     NaN  
15      0     NaN  
16      0     NaN  
17      1     NaN  
18      1     NaN  
19      1     NaN

In [53]:
df_vehicules.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 93585 entries, 0 to 93584
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Num_Acc      93585 non-null  int64  
 1   id_vehicule  93585 non-null  object 
 2   num_veh      93585 non-null  object 
 3   senc         93585 non-null  int64  
 4   catv         93585 non-null  int64  
 5   obs          93585 non-null  int64  
 6   obsm         93585 non-null  int64  
 7   choc         93585 non-null  int64  
 8   manv         93585 non-null  int64  
 9   motor        93585 non-null  int64  
 10  occutc       838 non-null    float64
dtypes: float64(1), int64(8), object(2)
memory usage: 7.9+ MB


In [56]:
df_vehicules.describe()

Num_Acc          senc          catv           obs          obsm  \
count  9.358500e+04  93585.000000  93585.000000  93585.000000  93585.000000   
mean   2.023000e+11      1.593589     13.679949      1.050307      1.655297   
std    1.581615e+04      0.817015     14.636220      3.168680      1.258317   
min    2.023000e+11     -1.000000     -1.000000     -1.000000     -1.000000   
25%    2.023000e+11      1.000000      7.000000      0.000000      1.000000   
50%    2.023000e+11      1.000000      7.000000      0.000000      2.000000   
75%    2.023000e+11      2.000000     10.000000      0.000000      2.000000   
max    2.023001e+11      3.000000     99.000000     17.000000      9.000000   

               choc          manv         motor      occutc  
count  93585.000000  93585.000000  93585.000000  838.000000  
mean       2.877117      6.772122      1.304515    2.103819  
std        2.413490      7.792551      1.141443    4.372988  
min       -1.000000     -1.000000     -1.000000    0.000000  
25%        1.000000      1.000000      1.000000    1.000000  
50%        2.000000      2.000000      1.000000    1.000000  
75%        4.000000     14.000000      1.000000    1.000000  
max        9.000000     26.000000      6.000000   65.000000

In [58]:
df_vehicules.isnull().sum()

Num_Acc            0
id_vehicule        0
num_veh            0
senc               0
catv               0
obs                0
obsm               0
choc               0
manv               0
motor              0
occutc         92747
dtype: int64

Jeu de données pour les véhicules de 93585 lignes pour 11 colonnes.
Un même accident ("Num_Acc") peut concerner plusieurs véhicules ("id_vehicule" et "num_veh") => on peut supprimer "num_veh"

### Suppression des colonnes inutiles


In [62]:
df_vehicules.drop(['num_veh'], axis = 1, inplace = True)
df_vehicules.drop(['occutc'], axis = 1, inplace = True)  #car renseignée qu'à 0.9%

In [64]:
df_vehicules.head()

Num_Acc  id_vehicule  senc  catv  obs  obsm  choc  manv  motor
0  202300000001  155 680 557     1    30    0     0     5     1      1
1  202300000002  155 680 556     2     7    0     1     1     1      1
2  202300000003  155 680 554     1     2    0     2     1    16      1
3  202300000003  155 680 555     2     7    0     2     2    15      1
4  202300000004  155 680 551     1     7    0     2     9     2      4

### Observation des variables

In [68]:
# Sens de circulation
df_vehicules['senc'].value_counts(normalize = True)

senc
 1    0.447112
 2    0.341957
 3    0.154555
 0    0.055276
-1    0.001101
Name: proportion, dtype: float64

In [72]:
# Suppression de la colonne car non pertinente (besoin d'infos métier)
df_vehicules.drop('senc', axis = 1, inplace = True)

In [78]:
# Catégorie du véhicule
df_vehicules['catv'].value_counts(normalize = True)

catv
 7     0.583641
 33    0.082150
 10    0.073228
 1     0.054667
 2     0.035326
 50    0.025592
 30    0.023679
 32    0.019266
 31    0.018796
 15    0.009991
 34    0.009542
 80    0.008142
 14    0.007651
 37    0.007213
 43    0.006604
 17    0.005802
 3     0.005150
 99    0.004199
 13    0.004018
 0     0.002768
 21    0.002703
 60    0.002671
 36    0.001923
 38    0.001742
 40    0.001175
 16    0.000470
 20    0.000449
 39    0.000417
 42    0.000395
 35    0.000374
 41    0.000224
-1     0.000032
Name: proportion, dtype: float64

In [19]:
# Valeurs de la variable doivent être comprises entre 00 et 99
# Suppression des lignes avec la valeur -1

In [80]:
df_vehicules = df_vehicules[df_vehicules['catv'] != -1]

In [82]:
# Reduction des classes en 1 seule quand <1%
df_vehicules['catv'] = df_vehicules['catv'].replace([15, 37, 34, 14, 80, 43, 17, 3, 13, 38, 21, 60, 0, 36, 40, 20, 16, 39, 35, 42, 41], 99)

C:\Users\peggy\AppData\Local\Temp\ipykernel_21320\2410329150.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_vehicules['catv'] = df_vehicules['catv'].replace([15, 37, 34, 14, 80, 43, 17, 3, 13, 38, 21, 60, 0, 36, 40, 20, 16, 39, 35, 42, 41], 99)


In [88]:
# Création de la classe 22 pour les "deux-roues motorisés" => regroupement de 02 – Cyclomoteur <50cm3, 33 – Motocyclette > 125 cm3, 50 – EDP à moteur, 
# 30 – Scooter < 50 cm3, 32 – Scooter > 50 cm3 et <= 125 cm3, 31 – Motocyclette > 50 cm3 et <= 125 cm3
df_vehicules['catv'] = df_vehicules['catv'].replace([2, 33, 50, 30, 31, 32], 22)

C:\Users\peggy\AppData\Local\Temp\ipykernel_21320\2006073633.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_vehicules['catv'] = df_vehicules['catv'].replace([2, 33, 50, 30, 31, 32], 22)


In [90]:
df_vehicules['catv'].value_counts(normalize = True)

catv
7     0.583659
22    0.204815
99    0.083627
10    0.073230
1     0.054669
Name: proportion, dtype: float64

In [94]:
# Obstacle fixe heurté
df_vehicules['obs'].value_counts(normalize = True)

obs
 0     0.849448
 1     0.023872
 13    0.018348
 2     0.015441
 4     0.014928
 3     0.013592
 8     0.011765
 6     0.011658
 12    0.008901
 14    0.007566
 15    0.005503
 9     0.004969
 16    0.004691
 7     0.002928
 5     0.001891
 11    0.001624
 10    0.001207
 17    0.001069
-1     0.000598
Name: proportion, dtype: float64

In [96]:
# Suppression de la colonne car non pertinente (pas d'obstacle dans 88% des cas)
df_vehicules.drop('obs', axis = 1, inplace = True)

C:\Users\peggy\AppData\Local\Temp\ipykernel_21320\1570949226.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_vehicules.drop('obs', axis = 1, inplace = True)


In [98]:
# Obstacle mobile heurté
df_vehicules['obsm'].value_counts(normalize = True)

obsm
 2    0.699707
 0    0.189780
 1    0.090167
 9    0.016616
 6    0.001432
 4    0.000994
 5    0.000823
-1    0.000481
Name: proportion, dtype: float64

In [100]:
# Suppression de la colonne car non pertinente 
df_vehicules.drop('obsm', axis = 1, inplace = True)

C:\Users\peggy\AppData\Local\Temp\ipykernel_21320\382276348.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_vehicules.drop('obsm', axis = 1, inplace = True)


In [104]:
# Point de choc
df_vehicules['choc'].value_counts(normalize = True)

choc
 1    0.364044
 3    0.148714
 2    0.122940
 4    0.096247
 8    0.069949
 0    0.062961
 7    0.057639
 6    0.034772
 5    0.028510
 9    0.013614
-1    0.000609
Name: proportion, dtype: float64

In [106]:
# Création de classes
df_vehicules['choc'] = df_vehicules['choc'].replace([2, 3], 1)
df_vehicules['choc'] = df_vehicules['choc'].replace([5, 6], 4)
df_vehicules['choc'] = df_vehicules['choc'].replace(8, 7)

C:\Users\peggy\AppData\Local\Temp\ipykernel_21320\1509137362.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_vehicules['choc'] = df_vehicules['choc'].replace([2, 3], 1)
C:\Users\peggy\AppData\Local\Temp\ipykernel_21320\1509137362.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_vehicules['choc'] = df_vehicules['choc'].replace([5, 6], 4)
C:\Users\peggy\AppData\Local\Temp\ipykernel_21320\1509137362.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
T

In [108]:
# Suppression des -1
df_vehicules = df_vehicules[df_vehicules['choc']!= -1]

In [110]:
# Manoeuvre principale avant l'accident 
df_vehicules['manv'].value_counts(normalize = True)

manv
 1     0.416594
 2     0.113884
 15    0.078888
 0     0.066763
 13    0.044459
 17    0.031403
 9     0.028025
 16    0.026164
 26    0.025587
 19    0.023951
 23    0.023021
 14    0.019963
 21    0.014884
 3     0.013633
 5     0.013472
 11    0.012446
 12    0.011366
 10    0.006640
 20    0.005688
 4     0.004833
 18    0.004608
 24    0.003849
 22    0.003079
 25    0.002951
 6     0.001786
 7     0.001647
 8     0.000235
-1     0.000182
Name: proportion, dtype: float64

In [112]:
# Suppression de la colonne car trop de classe difficile à réunir 
df_vehicules.drop('manv', axis = 1, inplace = True)

In [116]:
# Type de motorisation du véhicule
df_vehicules['motor'].value_counts(normalize = True)

motor
 1    0.779834
 0    0.080535
 5    0.057161
 3    0.053248
 2    0.020497
 6    0.005934
-1    0.001615
 4    0.001176
Name: proportion, dtype: float64

In [118]:
# Création de classes
df_vehicules['motor'] = df_vehicules['motor'].replace(-1, 0)
df_vehicules['motor'] = df_vehicules['motor'].replace(4, 6)
df_vehicules['motor'] = df_vehicules['motor'].replace(2, 3)

## Exportation du dataframe nettoyé

In [122]:
## Exportation du dataframe nettoyé
df_vehicules.to_csv('vehicules_clean.csv', index=False)